src/main/java/org/jabref/gui/citationkeypattern/GenerateCitationKeyAction.java

```diff
@@ -80,30 +83,51 @@ private void checkOverwriteKeysChosen() {
         }
     }
 
-    private void generateKeys() {
-        if (isCanceled) {
-            return;
-        }
-
-        stateManager.getActiveDatabase().ifPresent(databaseContext -> {
-            // generate the new citation keys for each entry
-            final NamedCompound compound = new NamedCompound(Localization.lang("Autogenerate citation keys"));
-            CitationKeyGenerator keyGenerator =
-                    new CitationKeyGenerator(databaseContext, Globals.prefs.getCitationKeyPatternPreferences());
-            for (BibEntry entry : entries) {
-                keyGenerator.generateAndSetKey(entry)
-                            .ifPresent(fieldChange -> compound.addEdit(new UndoableKeyChange(fieldChange)));
-            }
-            compound.end();
-
-            // register the undo event only if new citation keys were generated
-            if (compound.hasEdits()) {
-                frame.getUndoManager().addEdit(compound);
+    private BackgroundTask generateKeysInBackground() {
+        return new BackgroundTask<Void>() {
+
+            @Override
+            protected Void call() {
+
+                    if (isCanceled) {
+                        return null;
+                    }
+                    showToUser(true);
```

Comment:
There is this golden rule of concurrency in JavaFX, that you shouldn't update any UI-facing properties in background tasks (or only using `updateProgress` and `updateMessage`), see for example https://docs.oracle.com/javafx/2/api/javafx/concurrent/Task.html.

Thus, setting title and message should happen in the constructor and not in the call method.